sdk/messaging/azservicebus/internal/atom/models.go

```diff
@@ -0,0 +1,230 @@
+		DeadLetteringOnFilterEvaluationExceptions *bool                   `xml:"DeadLetteringOnFilterEvaluationExceptions,omitempty"`
+		DefaultRuleDescription                    *DefaultRuleDescription `xml:"DefaultRuleDescription,omitempty"`
+		MaxDeliveryCount                          *int32                  `xml:"MaxDeliveryCount,omitempty"`        // MaxDeliveryCount - The maximum delivery count. A message is automatically deadlettered after this number of deliveries. default value is 10.
+		MessageCount                              *int64                  `xml:"MessageCount,omitempty"`            // MessageCount - The number of messages in the queue.
+		EnableBatchedOperations                   *bool                   `xml:"EnableBatchedOperations,omitempty"` // EnableBatchedOperations - Value that indicates whether server-side batched operations are enabled.
+		Status                                    *EntityStatus           `xml:"Status,omitempty"`
+		ForwardTo                                 *string                 `xml:"ForwardTo,omitempty"` // ForwardTo - absolute URI of the entity to forward messages
+		UserMetadata                              *string                 `xml:"UserMetadata,omitempty"`
+		ForwardDeadLetteredMessagesTo             *string                 `xml:"ForwardDeadLetteredMessagesTo,omitempty"` // ForwardDeadLetteredMessagesTo - absolute URI of the entity to forward dead letter messages
+		AutoDeleteOnIdle                          *string                 `xml:"AutoDeleteOnIdle,omitempty"`
+		CreatedAt                                 *date.Time              `xml:"CreatedAt,omitempty"`
+		UpdatedAt                                 *date.Time              `xml:"UpdatedAt,omitempty"`
+		AccessedAt                                *date.Time              `xml:"AccessedAt,omitempty"`
+		CountDetails                              *CountDetails           `xml:"CountDetails,omitempty"`
+	}
+
+	// SubscriptionEntity is the Azure Service Bus description of a topic Subscription for management activities
+	SubscriptionEntity struct {
+		*SubscriptionDescription
+		*Entity
+	}
+
+	// SubscriptionFeed is a specialized feed containing Topic Subscriptions
+	SubscriptionFeed struct {
+		*Feed
+		Entries []SubscriptionEnvelope `xml:"entry"`
+	}
+
+	// subscriptionEntryContent is a specialized Topic feed Subscription
+	SubscriptionEnvelope struct {
+		*Entry
+		Content *subscriptionContent `xml:"content"`
+	}
+
+	// subscriptionContent is a specialized Subscription body for an Atom entry
+	subscriptionContent struct {
+		XMLName                 xml.Name                `xml:"content"`
+		Type                    string                  `xml:"type,attr"`
+		SubscriptionDescription SubscriptionDescription `xml:"SubscriptionDescription"`
+	}
+)
+
+type (
+	// Entity is represents the most basic form of an Azure Service Bus entity.
+	Entity struct {
```

Comment:
Does this need to be in a separate `type` declaration because it's referenced in the one above?

Reply:
You're so charitable. Just a silly oversight (now corrected).